plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -2261,6 +2261,43 @@ public static function get_install_base_url() {
 
 		return $woo_com_base_url . 'auto-install-init/';
 	}
+
+	/**
+	 * Retrieve notice for connected store.
+	 *
+	 * @return array An array containing notice data.
+	 */
+	public static function get_notices() {
+		$cache_key = '_woocommerce_helper_notices';
+		$cached_data = get_transient( $cache_key );
+
+		if ( false !== $cached_data ) {
+			return $cached_data;
+		}
+
+		// Fetch notice data for connected store.
+		$request = WC_Helper_API::get(
+			'notices'
+			,
+			[
+				'authenticated' => true,
+			]
+		);
+
+		if ( 200 !== wp_remote_retrieve_response_code( $request ) ) {
+			set_transient( $cache_key, [], 15 * MINUTE_IN_SECONDS );
+			return array();
+		}
+
+		$data = json_decode( wp_remote_retrieve_body( $request ), true );
+
+		if ( empty( $data ) || ! is_array( $data ) ) {
+			$data = array();
+		}
+
+		set_transient( $cache_key, $data, 12 * HOUR_IN_SECONDS );
```

Comment:
This feels too long, `/subscriptions` is cached for 1h.

Reply:
Once a customer added a payment method in WCCOM, we will still show the payment method not available notice for up to 12 hours in this case. IMHO we should put this cache duration shorter

Reply:
Updated to 1h in https://github.com/woocommerce/woocommerce/pull/47141/commits/d4190c98ed8929626ec0f04169d31170f4f36e47